Source/ThirdParty/ANGLE/src/libANGLE/renderer/metal/mtl_state_cache.mm

```diff
@@ -943,6 +943,60 @@ void ToObjC(const RenderPassStencilAttachmentDesc &desc,
     return re.first->second;
 }
 
+static bool ValidateRenderPipelineState(const MTLRenderPipelineDescriptor *descriptor,
+                                        ContextMtl *context,
+                                        const mtl::ContextDevice &device)
+{
+    // Ensure there is at least one valid render target.
+    bool hasValidRenderTarget = false;
+
+    for (NSUInteger i = 0;
+         !hasValidRenderTarget && i < GetMaxNumberOfRenderTargetsForDevice(device); ++i)
+    {
+        auto colorAttachment = descriptor.colorAttachments[i];
+        if (colorAttachment && colorAttachment.pixelFormat != MTLPixelFormatInvalid)
+        {
+            hasValidRenderTarget = true;
```

Comment:
Could also break here and avoid the test of hasValidRenderTarget in the loop header.